msal/application.py

```diff
@@ -1357,13 +1357,14 @@ def _acquire_token_silent_from_cache_and_possibly_refresh_it(
             key_id = kwargs.get("data", {}).get("key_id")
             if key_id:  # Some token types (SSH-certs, POP) are bound to a key
                 query["key_id"] = key_id
-            matches = self.token_cache.find(
-                self.token_cache.CredentialType.ACCESS_TOKEN,
-                target=scopes,
-                query=query)
             now = time.time()
             refresh_reason = msal.telemetry.AT_ABSENT
-            for entry in matches:
+            for entry in self.token_cache._find(  # It returns a generator
```

Comment:
Changing `token_cache.find` to `token_cache._find` makes this line no longer call the `find` method on `msal.token_cache.TokenCache`'s subclasses, e.g. `msal_extensions.token_cache.PersistedTokenCache.find`, but directly `msal.token_cache.TokenCache._find`.

This causes all logic in `msal_extensions.token_cache.PersistedTokenCache.find` being skipped:

https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/a10d0929b53d5397a3d5883e97b6925de0252e2b/msal_extensions/token_cache.py#L72-L87

This method contains 2 important functionalities:

1. Load token cache from a file
2. Handle concurrent situation

When a `PersistedTokenCache` is provided to `ConfidentialClientApplication`, as token cache is no longer loaded from a file, `self._cache` becomes an empty dict when this line is called:

https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/84bdfabfc3cbd73bca485b2420fcb7ccc01191d0/msal/token_cache.py#L118

resulting in the token cache being bypassed.

`PublicClientApplication` is not affected, as `application.ClientApplication._find_msal_accounts` still uses the old `find`.

https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/52b1fc5a442ff5dd33f48ce717f1032c8002ea9e/msal/application.py#L1111